driver-core/src/main/com/mongodb/internal/connection/InternalStreamConnection.java

```diff
@@ -349,9 +349,6 @@ public <T> T sendAndReceive(final CommandMessage message, final Decoder<T> decod
         CommandEventSender commandEventSender;
 
         try (ByteBufferBsonOutput bsonOutput = new ByteBufferBsonOutput(this)) {
-            Timeout.onExistsAndExpired(operationContext.getTimeoutContext().timeoutIncludingRoundTrip(), () -> {
```

Comment:
According to the specification: 

> After wire message construction, drivers MUST check for timeout before writing the message to the server.


I have removed this check from its previous position as it was occurring before the wire message was constructed, not after, which is contrary to the spec. We still perform this necessary timeout check later in the code within the trySendMessage method, just before writing the message to the socket.